### ISSUE TITLE
[OSM] Insecure HTTP API / or HTTPS?

### ISSUE THREAD

**Issue by rugk** [osmandapp/Osmand#3191]
In the code I saw you [sometimes](https://github.com/osmandapp/Osmand/blob/73dcab7d0b944e54a7fb08ee0f323e9588a2dfdb/OsmAnd/src/net/osmand/plus/osmedit/OsmBugsRemoteUtil.java#L35) [use](https://github.com/osmandapp/Osmand/blob/73dcab7d0b944e54a7fb08ee0f323e9588a2dfdb/OsmAnd/src/net/osmand/plus/osmedit/OsmBugsRemoteUtil.java#L44) [the](https://github.com/osmandapp/Osmand/blob/f4927f44e1871eefc63a9a58161d156fb14f7648/OsmAnd-java/src/net/osmand/util/GeoPointParserUtil.java#L266) [HTTP](https://github.com/osmandapp/Osmand/blob/73dcab7d0b944e54a7fb08ee0f323e9588a2dfdb/OsmAnd/src/net/osmand/plus/osmedit/OpenstreetmapRemoteUtil.java#L81) API.

So as I did look in your FAQ, but I cannot find clear information abut when HTTP is used and when HTTPS.

1. The most important question first: (When) Is any location information send over HTTP?
2. (When) Is the login data (for OSM) send over HTTP?
3. What other requests to OpenStreetMap are unencrypted?
4. What requests are encrypted?

You may thing about adding this information to the FAQ or the app directly.

In any case you should however try use HTTPS every time when it is possible. the OSM site is available in the HTTPS version for years. You might check the links I have above.

Additionally the OSM module in OSMand has the URL `http://openstreetmap.org` in it's description shown to the user. It is not really nice to see that `http://` there, as this might make users think data is send over (unencrypted) HTTP connections. I'd suggest you to either change this to `https://` or just leave the protocol out. 

Finally I of course hope you verify the ssl/tls certificates correctly as this is a mayor problem on Android.

So to summarize:
* [ ] We need more information about when HTTPS is used and when not.
* [ ] When HTTPS is not used (yet), try to use it.
* [ ] Do not use `http://openstreetmap.org` in the description.

**Comment by vshcherb** [comment]
HTTP is a dead option for 90% of devices, as you see in the code 		if (deviceApiVersion >= android.os.Build.VERSION_CODES.GINGERBREAD). So for devices since 3.0 we use only https

**Comment by rugk** [comment]
Please reopen this issue and again look at the lines I linked. Only 2 or 3 have an if clause.

Also the UI change has not been fixed yet.

**Comment by sonora** [comment]
- The one line you link without the if clause points to our geocoding parser :)
- I have changed the http:// to https:// in the Plugin description, for what it's worth, although that reference only points you to "where to get more information"

**Comment by rugk** [comment]
Okay, nice.
